src/main/java/online/book/store/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package online.book.store.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import online.book.store.dto.user.request.UserRegistrationRequestDto;
+import online.book.store.dto.user.response.UserResponseDto;
+import online.book.store.exception.EmailExistException;
+import online.book.store.mapper.UserMapper;
+import online.book.store.model.User;
+import online.book.store.repository.UserRepository;
+import online.book.store.service.UserService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto userRequestDto) {
+        if (userRepository.findByEmail(userRequestDto.getEmail()).isPresent()) {
```

Comment:
```suggestion
        if (userRepo.existsByEmail(request.email())) {
```
It's better to create existsByEmail method to avoid calling isPresent()